packages/persist/lib/records.ts

```diff
@@ -119,17 +117,30 @@ export async function persistRecords({
             void logCtx.error(`Found duplicate key '${nonUniqueKey}' for model ${baseModel}. The record was ignored.`);
         }
 
-        const total = summary.addedKeys.length + summary.updatedKeys.length + (summary.deletedKeys?.length || 0);
-        void logCtx.info(`Successfully batched ${total} record${total > 1 ? 's' : ''}`, {
+        await updateSyncJobResult(syncJobId, updatedResults, baseModel);
+
+        const allModifiedKeys = new Set([...summary.addedKeys, ...summary.updatedKeys, ...(summary.deletedKeys || [])]);
+        void logCtx.info(`Successfully batched ${allModifiedKeys.size} record${allModifiedKeys.size > 1 ? 's' : ''}`, {
             persistType,
             updatedResults
         });
-        await updateSyncJobResult(syncJobId, updatedResults, baseModel);
 
-        metrics.increment(metrics.Types.PERSIST_RECORDS_COUNT, records.length, { accountId });
+        const recordsSizeInBytes = recordsData.reduce((acc, record) => {
+            if (allModifiedKeys.has(record.id)) {
+                return acc + Buffer.byteLength(JSON.stringify(record), 'utf8');
+            }
+            return acc;
+        }, 0);
+
+        metrics.increment(metrics.Types.PERSIST_RECORDS_COUNT, allModifiedKeys.size, { accountId });
         metrics.increment(metrics.Types.PERSIST_RECORDS_SIZE_IN_BYTES, recordsSizeInBytes, { accountId });
 
+        span.addTags({
+            'records.count': allModifiedKeys.size,
```

Comment:
not sure I agree with this, or at least I would log both count but having 0 because nothing was modified is definitely not helping for debugging

Reply:
Good idea. let's have 2 tags:`records.modified.count` and `records.in.count`?